srml/contracts/src/wasm/mod.rs

```diff
@@ -1634,4 +1662,103 @@ mod tests {
 		assert_eq!(output, ExecReturnValue { status: 17, data: hex!("5566778899").to_vec() });
 		assert!(!output.is_success());
 	}
+
+	const CODE_GET_RUNTIME_STORAGE: &str = r#"
+(module
+	(import "env" "ext_get_runtime_storage"
+		(func $ext_get_runtime_storage (param i32 i32) (result i32))
+	)
+	(import "env" "ext_scratch_size" (func $ext_scratch_size (result i32)))
+	(import "env" "ext_scratch_read" (func $ext_scratch_read (param i32 i32 i32)))
+	(import "env" "ext_scratch_write" (func $ext_scratch_write (param i32 i32)))
+	(import "env" "memory" (memory 1 1))
+
+	(func (export "deploy"))
+
+	(func $assert (param i32)
+		(block $ok
+			(br_if $ok
+				(get_local 0)
+			)
+			(unreachable)
+		)
+	)
+
+	(func $call (export "call")
+		;; Load runtime storage for the first key and assert that it exists.
+		(call $assert
+			(i32.eq
+				(call $ext_get_runtime_storage
+					(i32.const 16)
+					(i32.const 4)
+				)
+				(i32.const 0)
+			)
+		)
+
+		;; assert $ext_scratch_size == 4
+		(call $assert
+			(i32.eq
+				(call $ext_scratch_size)
+				(i32.const 4)
+			)
+		)
+
+		;; copy contents of the scratch buffer into the contract's memory.
+		(call $ext_scratch_read
+			(i32.const 4)		;; Pointer in memory to the place where to copy.
+			(i32.const 0)		;; Offset from the start of the scratch buffer.
+			(i32.const 4)		;; Count of bytes to copy.
+		)
+
+		;; assert that contents of the buffer is equal to the i32 value of 0x14144020.
+		(call $assert
+			(i32.eq
+				(i32.load
+					(i32.const 4)
+				)
+				(i32.const 0x14144020)
+			)
+		)
+
+		;; Load the second key and assert that it doesn't exist.
+		(call $assert
+			(i32.eq
+				(call $ext_get_runtime_storage
+					(i32.const 20)
+					(i32.const 4)
+				)
+				(i32.const 1)
+			)
+		)
+	)
+
+	;; The first key, 4 bytes long.
+	(data (i32.const 16) "\01\02\03\04")
+	;; The second key, 4 bytes long.
+	(data (i32.const 20) "\02\03\04\05")
+)
+"#;
+
+	#[test]
+	fn get_runtime_storage() {
+		let mut gas_meter = GasMeter::with_limit(50_000, 1);
+		let mock_ext = MockExt::default();
+
+		// "\01\02\03\04" - Some(0x14144020)
+		// "\02\03\04\05" - None
+		*mock_ext.runtime_storage_keys.borrow_mut() = [
+			([1, 2, 3, 4].to_vec(), Some(0x14144020u32.to_le_bytes().to_vec())),
+			([2, 3, 4, 5].to_vec().to_vec(), None),
+		]
+		.iter()
+		.cloned()
+		.collect();
+		let _ = execute(
```

Comment:
Why `let` here required if there is also an `unwrap`?

Reply:
`let` here is not required. Note all other usages of this pattern - it is deliberate stylistic choise to note that there is a result. Don't feel strongly about this one though, and can be removed. Although also, not within this PR.

Reply:
It confused me so maybe it confuses others as well. Agree that we shouldn't do that in this PR if it is a pattern but you might want to think about removing it because it might confuse others.